docs/api/StreamSend.md

```diff
@@ -21,15 +21,73 @@ QUIC_STATUS
 
 # Parameters
 
-**TODO**
+`Stream`
+
+The valid handle to an open stream object.
+
+`Buffers`
+
+An array of `QUIC_BUFFER` structs that each contain a pointer and length to app data to send on the stream. This may be `NULL` **only** if `BufferCount` is zero.
+
+`BufferCount`
+
+The number of `QUIC_BUFFER` structs in the `Buffers` array. This may be zero.
+
+`Flags`
+
+The set of flags the controls the behavior of `StreamSend`:
+
+Value | Meaning
+--- | ---
+**QUIC_SEND_FLAG_NONE**<br>0 | No special behavior. Data is not allowed in 0-RTT by default.
+**QUIC_SEND_FLAG_ALLOW_0_RTT**<br>1 | Indicates that the data is allowed to be sent in 0-RTT (if available). Makes no guarantee the data will be sent in 0-RTT. Additionally, even if 0-RTT keys are available the data may end up being sent in 1-RTT for multiple reasons.
+**QUIC_SEND_FLAG_START**<br>2 | Indicates that the stream should asynchronously start (equivalent to calling [StreamStart](StreamStart.md) with the `QUIC_STREAM_START_FLAG_ASYNC` flag).
```

Comment:
I don't understand this one...

Reply:
"Starting a stream" essentially means "start sending stuff on the wire". This flag just allows you to condense a StreamStart and StreamSend into a single StreamSend call. There is an example below that I think clarifies more. Does that answer your question?

Reply:
Ahhhh, I get it now. It's just so I can avoid the StreamStart call? Is there a perf advantage to that, I assume?

Does this imply QUIC_STREAM_START_FLAG_IMMEDIATE or not? (related to below question I suspect)

I get the purpose of this now, but I think the doc could be clearer... Maybe something like "Indicates the stream should also be started asynchronously.... This allows the Start and Send operations to be done in a single call"

Reply:
Yes, this is a performance feature. The `QUIC_STREAM_START_FLAG_IMMEDIATE` flag is not applicable in this scenario because you already have data to send so it will go out. That flag was just about forcing a send even when there was no data yet.

Reply:
To be clear -- if I call StreamStart w/o QUIC_STREAM_START_FLAG_IMMEDIATE, then call StreamSend with 0 bytes, will that be effectively the same as calling StreamStart with QUIC_STREAM_START_FLAG_IMMEDIATE? I would assume so, just wanted to confirm....

Reply:
I would expect so, yes, but I need to double check the code to see if that actually works. I don't think it has ever been tried.

Reply:
Created https://github.com/microsoft/msquic/pull/1608 to handle this case.